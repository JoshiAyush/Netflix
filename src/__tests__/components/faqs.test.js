import React from "react";
import { render, fireEvent } from "@testing-library/react";

import faqsData from "../../fixtures/faqs.json";

import { Faqs } from "../../containers/index.js";

describe("<Accordion />", () => {
    it("renders the <Accordion /> with populated data", () => {
        const { container, getByText } = render(
            <Faqs />
        );

        expect(getByText("Frequently Asked Question")).toBeTruthy();
        expect(getByText("What is Netflix?")).toBeTruthy();
        expect(getByText("How much does Netflix cost?")).toBeTruthy();
        expect(getByText("Where can I watch?")).toBeTruthy();
        expect(getByText("How do I cancel?")).toBeTruthy();
        expect(getByText("What can I watch on Netflix?")).toBeTruthy();

        expect(container.firstChild).toMatchSnapshot();
    });

    it("opens and closes the <Accordion /> component", () => {
        const { container, queryByText } = render(
            <Faqs />
        );

        const whatIsNetflixBodyText = faqsData.filter(item => item.header == "What is Netflix?")[0].body;

        expect(queryByText(whatIsNetflixBodyText)).toBeFalsy();
        fireEvent.click(queryByText("What is Netflix?"));
        expect(queryByText(whatIsNetflixBodyText)).toBeTruthy();
        fireEvent.click(queryByText("What is Netflix?"));
        expect(queryByText(whatIsNetflixBodyText)).toBeFalsy();

        expect(container.firstChild).toMatchSnapshot();
    });
});
