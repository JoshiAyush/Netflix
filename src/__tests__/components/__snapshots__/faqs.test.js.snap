// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> renders the <Accordion /> with populated data 1`] = `
<div
  class="style__Container-uywbgw-0"
>
  <div
    class="style__Container-o549vb-0 dseBTj"
  >
    <div
      class="style__Inner-o549vb-5 fCAWwI"
    >
      <div
        class="style__Title-o549vb-6 bIaLso"
      >
        Frequently Asked Question
      </div>
      <div
        class="style__Item-o549vb-1 jNuosP"
      >
        <div
          class="style__Header-o549vb-2 dLNBwC"
        >
          What is Netflix?
          <img
            alt="Open"
            class="style__Image-o549vb-7"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="style__Item-o549vb-1 jNuosP"
      >
        <div
          class="style__Header-o549vb-2 dLNBwC"
        >
          How much does Netflix cost?
          <img
            alt="Open"
            class="style__Image-o549vb-7"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="style__Item-o549vb-1 jNuosP"
      >
        <div
          class="style__Header-o549vb-2 dLNBwC"
        >
          Where can I watch?
          <img
            alt="Open"
            class="style__Image-o549vb-7"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="style__Item-o549vb-1 jNuosP"
      >
        <div
          class="style__Header-o549vb-2 dLNBwC"
        >
          How do I cancel?
          <img
            alt="Open"
            class="style__Image-o549vb-7"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="style__Item-o549vb-1 jNuosP"
      >
        <div
          class="style__Header-o549vb-2 dLNBwC"
        >
          What can I watch on Netflix?
          <img
            alt="Open"
            class="style__Image-o549vb-7"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="style__Item-o549vb-1 jNuosP"
      />
      <div
        class="style__Container-aw8769-0 emHGMr"
      >
        <input
          class="style__Input-aw8769-1 jcPedt"
          placeholder="Email address"
        />
        <button
          class="style__Button-aw8769-2 kiOXdQ"
        >
          Try it now
           
          <img
            alt="Try Now"
            src="/images/icons/chevron-right.png"
          />
        </button>
        <p
          class="style__Text-aw8769-3 dicfXI"
        >
          Ready to watch? Enter your email to create or restart your membership
        </p>
      </div>
    </div>
  </div>
</div>
`;
